### ISSUE TITLE
[Bug]: Plugin not picking up devices identified using "Label" feature

### ISSUE THREAD

**Issue by jvmahon** [t0bst4r/matterbridge-home-assistant-addon#103]
### What happened?

Thank you for adding in the new feature to select devices using labels.  Unfortunately, the plugin does not seem to be picking up my labeled devices.  My configuration is very simple - include devices with the label "matterbridge".

I then have two Z-wave devices with that label:
![image](https://github.com/user-attachments/assets/adf1961e-a644-49f2-b0d9-53edf1cec1be)

I have the Matterbridge logging level set to "Info" - nothing relevant shows up.

![image](https://github.com/user-attachments/assets/149e0c7f-db5a-4048-9e71-ee42ef505e88)

I have tried to change the Matterbridge logging level to "Debug" but when I do, the plugin fails to restart (it just gets stuck on "Loading Settings".

![image](https://github.com/user-attachments/assets/d554476c-1c7d-4903-9067-64ffb18f495f)

### Your configuration

![image](https://github.com/user-attachments/assets/0d049b98-dde0-4b73-9bc0-f06d6de0c4fd)

### Relevant log output

```shell
#############################
CURRENT CONFIGURATION:
{
  "homeAssistant": {
    "url": "http://supervisor/core",
    "accessToken": "***",
    "matcher": {
      "includeDomains": [],
      "excludeDomains": [],
      "includePatterns": [],
      "excludePatterns": [],
      "includeLabels": [
        "matterbridge"
      ],
      "excludeLabels": [],
      "includePlatforms": [],
      "excludePlatforms": []
    }
  }
}
#############################
```

### Version

2.1.1-99

**Comment by t0bst4r** [comment]
First I'd recommend changing the log level in the Home Assistant AddOn configuration. With that you''ll get even more details.
Anyway you won't find any more details regarding your problem. It will only say that your entity is ignored, but it won't tell you why.

I'll try to reproduce it and then post an update here.

**Comment by jvmahon** [comment]
Solved - when I tried the template, it showed no labels.  

I had applied the label at the "device" level, and it looks like it has to be applied at the entity level instead (which makes sense, particularly if you have multi-endpoint devices). After correcting that, I tried the template again and it does show "matterbridge" as the entity's label and when I restarted the plugin, it worked.

Consider adding a brief explanation on the Configurations page "label must be applied to a HA entity on the Entities tab, not to a device on the Devices tab."

Thank you!
